xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -14883,6 +14883,11 @@ glusterd_check_brick_order(dict_t *dict, char *err_str, int32_t type,
     pre_list = MALLOC(sizeof(addrinfo_list_t));
     pre_list->info = NULL;
     CDS_INIT_LIST_HEAD(&pre_list->list);
+    if (pre_list == NULL) {
+        gf_msg(this->name, GF_LOG_ERROR, ENOMEM, GD_MSG_NO_MEMORY,
+               "failed to allocate memory");
+	goto out;
+    }
```

Comment:
Shouldn't this check move to line:1484?
If MALLOC fails, then we are dereferencing NULL at lines 1484 and 1485.